cpp/src/arrow/compute/kernels/aggregate_basic_avx2.cc

```diff
@@ -15,7 +15,28 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#include <cassert>
```

Comment:
Not sure why all the additional inclusions here.

Reply:
Because I include the `.cc` file in another `.cc` file and `#pragma once` apparently doesn't work in that case.

Reply:
Perhaps use an old-style `#ifdef`-based guard then?